### PR TITLE
Fix problem with Googletest's LIBRARY_OUTPUT_DIRECTORY

### PR DESCRIPTION
**Pull request by roehling**
The horribly maintained CMake files which are shipped with Googletest
override the library output directory to ${CMAKE_BINARY_DIR}/lib, which
points to the top-level build directory (it should have been
CMAKE_CURRENT_BINARY_DIR).

Unfortunately, that directory is used by setuptools as staging area for
Python modules, and `setup.py install` will happily install everything it
finds in the lib/ folder, including the libgtest.so binary. Among
others, this has bitten PR2/pr2_mechanism#343 and ros/geometry#223

This PR fixes the output location to something sane(ish).

### CODE REVIEW

**Comment by v4hn** [comment]
Thanks for looking into this Timo! You're proving the point that talking to colleagues about an issue at times resolves the issue :smile: :bouquet:

**Comment by jspricke** [comment]
Thanks! Please rebuild ros-noetic-controller-manager-tests, ros-noetic-pr2-controller-manager, ros-noetic-tf after releasing this.
Found using: `ros/pool/main/r$ for f in ros-noetic-*/*.deb; do dpkg-deb -c "$f" | grep -q "python3/dist-packages/libgtest.so" && echo "$f"; done`

**Comment by davefeilseifer** [comment]
was this rebuild done? it seems the most recent noetic sync did not fix this problem. Installing both ros-noetic-pr2-mechanism is still showing a conflict

**Comment by roehling** [comment]
@davefeilseifer I doubt it, the PR is still unmerged.

**Comment by davefeilseifer** [comment]
this is holding up prerelease testing for several other pr2 packages, is there anything that can be done to hurry this along?

**Comment by v4hn** [comment]
ping @mjcarroll @dirk-thomas

**Comment by dirk-thomas** [comment]
> ping

@v4hn I am not a maintainer of this repository anymore.

**Comment by peci1** [comment]
@tfoote @mjcarroll This has been holding the whole Moveit and dependent packages from Noetic release for more than a month...

**Review by mjcarroll — APPROVED** [review]

**Comment by fmauch** [comment]
Any chance this would get applied to melodic, as well? We've also stumbled upon this in our ros-melodic package builds on Archlinux (see https://github.com/ros-melodic-arch/ros-melodic-tf/issues/6)

**Pull request merged**
